### DISCUSSION TITLE
Generate id from Raven DB script

### DISCUSSION THREAD

**Discussion by agustinus010** [ravendb/ravendb#19108] [Q&A]
Hi all,

I need to create a patch script where I need to generate ID in the form of guid from the script
Would anyone know if there is built-in functionality or if there is other way to do so ?

thank you

**Comment by georgiosd** [comment]
Generate it from your code that sends the patch, and pass it as a patch
argument!

Georgios

**Reply by agustinus010** [reply]
Thanks Georgios,

afraid that might not work since that means I need to know how many id's I would need beforehand ?

**Reply by ppekrol** [reply]
There is no built-in method for this, you would have to write a piece of JS to generate them for you

**Reply by ayende** [reply]
Note that you can try doing this:

```
declare function uuidv4() {
    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'
        .replace(/[xy]/g, function (c) {
            const r = Math.random() * 16 | 0;
            const v = c == 'x' ? r : (r & 0x3 | 0x8);
            return v.toString(16);
        });
}
from "Fach"  as f
select {
    Uuid: uuidv4(),
    F: f
}
```

`Math.random` isn't _really_ that high quality, but should probably suffice.

Otherwise, calling `put()` on a document with no id will end up calling` Guid.NewGuid()` to generate the id.

All of that said, just use `prefix/` here, and RavenDB will generate an id for you. Why do you need a GUID like?
